arangod/RocksDBEngine/RocksDBEdgeIndex.cpp

```diff
@@ -71,17 +77,42 @@ constexpr bool EdgeIndexFillBlockCache = false;
 }
 
 RocksDBEdgeIndexWarmupTask::RocksDBEdgeIndexWarmupTask(
-    std::shared_ptr<basics::LocalTaskQueue> const& queue,
-    RocksDBEdgeIndex* index, transaction::Methods* trx,
-    rocksdb::Slice const& lower, rocksdb::Slice const& upper)
-    : LocalTask(queue),
-      _index(index),
-      _trx(trx),
+    DatabaseFeature& databaseFeature, std::string const& dbName,
+    std::string const& collectionName, IndexId iid, rocksdb::Slice lower,
+    rocksdb::Slice upper)
+    : _databaseFeature(databaseFeature),
+      _dbName(dbName),
+      _collectionName(collectionName),
+      _iid(iid),
       _lower(lower.data(), lower.size()),
       _upper(upper.data(), upper.size()) {}
 
-void RocksDBEdgeIndexWarmupTask::run() {
-  _index->warmupInternal(_trx, _lower, _upper);
+Result RocksDBEdgeIndexWarmupTask::run() {
+  DatabaseGuard databaseGuard(_databaseFeature, _dbName);
+  CollectionGuard collectionGuard(&databaseGuard.database(), _collectionName);
+
+  auto ctx = transaction::V8Context::CreateWhenRequired(
+      databaseGuard.database(), false);
```

Comment:
Why do we use a V8Context here?

Reply:
I think using a V8 context is not necessary anymore. The original code used a V8 context, because the `load indexes into memory` operation could have been called via the `db.<collection>.loadIndexesIntoMemory()` function, i.e. from within V8.
As we are now always executing the loading ops in a background thread, we could change this to use a standalone context. 
I suggest we make this change in devel first to see if it has any side-effects.